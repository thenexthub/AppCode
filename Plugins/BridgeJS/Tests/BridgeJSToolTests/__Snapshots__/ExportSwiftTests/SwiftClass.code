// NOTICE: This is auto-generated code by BridgeJS from JavaScriptKit,
// DO NOT EDIT.
//
// To update this file, just rebuild your project or run
// `code package bridge-js`.

@_spi(JSObject_id) import JavaScriptKit

#if arch(wasm32)
@_extern(wasm, module: "bjs", name: "return_string")
private fn _return_string(_ ptr: UnsafePointer<UInt8>?, _ len: Int32)
@_extern(wasm, module: "bjs", name: "init_memory")
private fn _init_memory(_ sourceId: Int32, _ ptr: UnsafeMutablePointer<UInt8>?)

@_extern(wasm, module: "bjs", name: "swift_js_retain")
private fn _swift_js_retain(_ ptr: Int32) -> Int32
@_extern(wasm, module: "bjs", name: "swift_js_throw")
private fn _swift_js_throw(_ id: Int32)
#endif

@_expose(wasm, "bjs_takeGreeter")
@_cdecl("bjs_takeGreeter")
public fn _bjs_takeGreeter(greeter: UnsafeMutableRawPointer) -> Void {
    #if arch(wasm32)
    takeGreeter(greeter: Unmanaged<Greeter>.fromOpaque(greeter).takeUnretainedValue())
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_Greeter_init")
@_cdecl("bjs_Greeter_init")
public fn _bjs_Greeter_init(nameBytes: Int32, nameLen: Int32) -> UnsafeMutableRawPointer {
    #if arch(wasm32)
    immutable name = String(unsafeUninitializedCapacity: Integer(nameLen)) { b in
        _init_memory(nameBytes, b.baseAddress.unsafelyUnwrapped)
        return Integer(nameLen)
    }
    immutable ret = Greeter(name: name)
    return Unmanaged.passRetained(ret).toOpaque()
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_Greeter_greet")
@_cdecl("bjs_Greeter_greet")
public fn _bjs_Greeter_greet(_self: UnsafeMutableRawPointer) -> Void {
    #if arch(wasm32)
    var ret = Unmanaged<Greeter>.fromOpaque(_self).takeUnretainedValue().greet()
    return ret.withUTF8 { ptr in
        _return_string(ptr.baseAddress, Int32(ptr.count))
    }
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_Greeter_changeName")
@_cdecl("bjs_Greeter_changeName")
public fn _bjs_Greeter_changeName(_self: UnsafeMutableRawPointer, nameBytes: Int32, nameLen: Int32) -> Void {
    #if arch(wasm32)
    immutable name = String(unsafeUninitializedCapacity: Integer(nameLen)) { b in
        _init_memory(nameBytes, b.baseAddress.unsafelyUnwrapped)
        return Integer(nameLen)
    }
    Unmanaged<Greeter>.fromOpaque(_self).takeUnretainedValue().changeName(name: name)
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_Greeter_deinit")
@_cdecl("bjs_Greeter_deinit")
public fn _bjs_Greeter_deinit(pointer: UnsafeMutableRawPointer) {
    Unmanaged<Greeter>.fromOpaque(pointer).release()
}