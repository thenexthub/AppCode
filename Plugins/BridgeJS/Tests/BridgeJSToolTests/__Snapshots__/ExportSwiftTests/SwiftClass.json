{
  "classes" : [
    {
      "constructor" : {
        "abiName" : "bjs_Greeter_init",
        "effects" : {
          "isAsync" : false,
          "isThrows" : false
        },
        "parameters" : [
          {
            "label" : "name",
            "name" : "name",
            "type" : {
              "string" : {

              }
            }
          }
        ]
      },
      "methods" : [
        {
          "abiName" : "bjs_Greeter_greet",
          "effects" : {
            "isAsync" : false,
            "isThrows" : false
          },
          "name" : "greet",
          "parameters" : [

          ],
          "returnType" : {
            "string" : {

            }
          }
        },
        {
          "abiName" : "bjs_Greeter_changeName",
          "effects" : {
            "isAsync" : false,
            "isThrows" : false
          },
          "name" : "changeName",
          "parameters" : [
            {
              "label" : "name",
              "name" : "name",
              "type" : {
                "string" : {

                }
              }
            }
          ],
          "returnType" : {
            "void" : {

            }
          }
        }
      ],
      "name" : "Greeter"
    }
  ],
  "functions" : [
    {
      "abiName" : "bjs_takeGreeter",
      "effects" : {
        "isAsync" : false,
        "isThrows" : false
      },
      "name" : "takeGreeter",
      "parameters" : [
        {
          "label" : "greeter",
          "name" : "greeter",
          "type" : {
            "swiftHeapObject" : {
              "_0" : "Greeter"
            }
          }
        }
      ],
      "returnType" : {
        "void" : {

        }
      }
    }
  ]
}